### ISSUE TITLE
Can't Install Minishift: Pre-flight check for persistent storage volume shows "Insufficient disk space"

### ISSUE THREAD

**Issue by karlnicholas** [minishift/minishift#1923]
* Minishift version: 1.11.0
  * Windows 10 Home - Version 1709 - Build 16299.125 (up to date)
  * VirtualBox - Version 5.2.6 r120293 (Qt5.6.2)

  1.  Download Minishift
  2. Run minishift start --vm-driver=virtualbox
It ran OK once I think last week, and worked all day, but then the problem happened again the next day.
This happened again just now:

```
C:\Users\karln>minishift start --vm-driver=virtualbox
-- Starting profile 'minishift'
-- Checking if requested hypervisor 'virtualbox' is supported on this platform ... OK
-- Checking if VirtualBox is installed ... OK
-- Checking the ISO URL ... OK
-- Starting local OpenShift cluster using 'virtualbox' hypervisor ...
-- Minishift VM will be configured with ...
   Memory:    2 GB
   vCPUs :    2
   Disk size: 20 GB

   Downloading ISO 'https://github.com/minishift/minishift-b2d-iso/releases/download/v1.2.0/minishift-b2d.iso'
 40.00 MiB / 40.00 MiB [===================================================================================] 100.00% 0s
-- Starting Minishift VM ............................ OK
-- Checking for IP address ... OK
-- Checking if external host is reachable from the Minishift VM ...
   Pinging 8.8.8.8 ... OK
-- Checking HTTP connectivity from the VM ...
   Retrieving http://minishift.io/index.html ... OK
-- Checking if persistent storage volume is mounted ... OK
-- Checking available disk space ... warning:
 used FAIL
   Insufficient disk space on the persistent storage volume

```

![virtualbox_minishift_18_01_2018_13_08_20](https://user-images.githubusercontent.com/2110283/35121583-becd51c8-fc50-11e7-8053-f9b918f6a895.png)

**Comment by gbraad** [comment]
Please check the documentation on how to handle this situation:

https://docs.openshift.org/latest/minishift/troubleshooting/troubleshooting-getting-started.html#persistent-storage-check

Depending on the hypervisor you need to either enlarge the disk, or choose to copy data out and recreate with a larger size. We are not able to perform the changing of the disk size.

**Comment by gbraad** [comment]
Marking as invalid, as this works 'as expected'. When the disk is full, unpredictible behaviour will occur, and with this check we want to inform the user beforehand.

We might want to include a link or reference to the troubleshooting segment in the error.

**Comment by budhrg** [comment]
> We might want to include a link or reference to the troubleshooting segment in the error.

Sure. Hence, changed the labels.

**Comment by karlnicholas** [comment]
Yea, I thought you were going to say something like that. The assumption that I haven't tried these things or that the openshift documentation addresses the issue is pretty much lacking. Please note the screen shot above where 450K is allocated out of 17 GB free. I don't think disk space is an issue. None the less, I tried it (again) with 50GB, but the same result ensues. I download minishift, I delete the .minishift directory from my home directory, and I type `minishift start --vm-driver=virstualbox`. You would think that would just work.

Exactly how much space do I need to allocate for a single instance of kunernetes? 

```
C:\Users\karln>minishift start --disk-size=50GB --vm-driver=virtualbox
-- Starting profile 'minishift'
-- Checking if requested hypervisor 'virtualbox' is supported on this platform ... OK
-- Checking if VirtualBox is installed ... OK
-- Checking the ISO URL ... OK
-- Starting local OpenShift cluster using 'virtualbox' hypervisor ...
-- Minishift VM will be configured with ...
   Memory:    2 GB
   vCPUs :    2
   Disk size: 50 GB

   Downloading ISO 'https://github.com/minishift/minishift-b2d-iso/releases/download/v1.2.0/minishift-b2d.iso'
 40.00 MiB / 40.00 MiB [===================================================================================] 100.00% 0s
-- Starting Minishift VM ............................... OK
-- Checking for IP address ... OK
-- Checking if external host is reachable from the Minishift VM ...
   Pinging 8.8.8.8 ... OK
-- Checking HTTP connectivity from the VM ...
   Retrieving http://minishift.io/index.html ... OK
-- Checking if persistent storage volume is mounted ... OK
-- Checking available disk space ... warning:
 used FAIL
   Insufficient disk space on the persistent storage volume
```

**Comment by karlnicholas** [comment]
Detailed log of disk check space:

> -- Checking available disk space ... (minishift) Calling .GetSSHHostname
> (minishift) Calling .GetSSHPort
> (minishift) Calling .GetSSHKeyPath
> (minishift) Calling .GetSSHKeyPath
> (minishift) Calling .GetSSHUsername
> Using SSH client type: external
> &{[-F /dev/null -o PasswordAuthentication=no -o StrictHostKeyChecking=no -o UserKnownHostsFile=/dev/null -o LogLevel=quiet -o ConnectionAttempts=3 -o ConnectTimeout=10 -o ControlMaster=no -o ControlPath=none docker@127.0.0.1 -o IdentitiesOnly=yes -i C:\Users\karln\.minishift\machines\minishift\id_rsa -p 65245] C:\Program Files\OpenShift\bin\ssh.exe <nil>}
> About to run SSH command:
> df -h /mnt/sda1 | awk 'FNR > 1 {print $2,$5}'
> SSH cmd err, output: <nil>: cygwin warning:
>   MS-DOS style path detected: C:\Users\karln\.minishift\machines\minishift\id_rsa
>   Preferred POSIX equivalent is: /cygdrive/c/Users/karln/.minishift/machines/minishift/id_rsa
>   CYGWIN environment variable option "nodosfilewarning" turns off this warning.
>   Consult the user's guide for more details about POSIX paths:
>     http://cygwin.com/cygwin-ug-net/using.html#using-pathnames
> 17.9G 0%
> 
> warning:
>  used FAIL
>    Insufficient disk space on the persistent storage volume

**Comment by karlnicholas** [comment]
Okay, I did two things and it worked

1) Set CYGWIN=nodosfilewarning in my environment
2) moved openshift/bin/ssh.exe to c:\users\karln (from c:\program files\) because maybe the space in the path name was breaking things: Got successful disk check and installation

```
-- Checking available disk space ... (minishift) Calling .GetSSHHostname
(minishift) Calling .GetSSHPort
(minishift) Calling .GetSSHKeyPath
(minishift) Calling .GetSSHKeyPath
(minishift) Calling .GetSSHUsername
Using SSH client type: external
&{[-F /dev/null -o PasswordAuthentication=no -o StrictHostKeyChecking=no -o UserKnownHostsFile=/dev/null -o LogLevel=quiet -o ConnectionAttempts=3 -o ConnectTimeout=10 -o ControlMaster=no -o ControlPath=none docker@127.0.0.1 -o IdentitiesOnly=yes -i C:\Users\karln\.minishift\machines\minishift\id_rsa -p 49174] C:\users\karln\OpenShift\bin\ssh.exe <nil>}
About to run SSH command:
df -h /mnt/sda1 | awk 'FNR > 1 {print $2,$5}'
SSH cmd err, output: <nil>: 17.9G 0%

0% used OK

```

**Comment by gbraad** [comment]
This is a new instance? Set the test to skip for now. Or can you try
starting with '--iso-url centos' ?

Something ain't right, since out of the box, with 20G it should surely
work. I use even less space for test instances.



-- 

   Gerard Braad | http://gbraad.nl
   [ Doing Open Source Matters ]

**Comment by gbraad** [comment]
Which ssh client did you install?

We have tested cygwin and this should be OK. Things like Gow is known to
fail. Also, openssh-win64 is OK

-- 

   Gerard Braad | http://gbraad.nl
   [ Doing Open Source Matters ]

**Comment by gbraad** [comment]
Ask as I not completely following point 2:

> openshift/bin/ssh.exe to c:\users\karln (from c:\program files)

openshift/bin/ssh.exe ?

-- 

   Gerard Braad | http://gbraad.nl
   [ Doing Open Source Matters ]

**Comment by karlnicholas** [comment]
the ssh.exe "installed" was from cwrsync which I added to the OC bin directory so that I could use rsync with OC to manage my permanent stores on openshift.  (https://www.itefix.net/content/cwrsync-free-edition). I just DL's those binaries and copied them into the openshift/bin directory. 

Actually, I have just DL'd the redhat Container Development Kit Version (CDK) so that I can use all this in DevStudio. I don't think I have ssh installed in Cygwin but maybe the cwrsync version uses that one. Personally, I'm good to go now and I probably saw it all working once when I initially installed the DevStio CDK Bundle which comes with cygwin and then changed things back to my own installation of Cygwin and got hung up on the SSH issue. Sorry for the confusion -- I presume another fix might be to have the Cygwin binaries in your path and be sure SSH is installed there. 

There is a another version of SSH I have which is Microsoft's beta SSH you can get by adding features in Windows/Setup but that's later in the PATH and so doesn't get executed. I didn't try it to see if it ran the minishift installation correctly.

**Comment by karlnicholas** [comment]
In terms of setting the test to skip I had tried that but still had problems. I presume again that wherever the installation was checking the results from the SSH I was using was failing and so still giving problems. 

Probably I'll "bite the bullet" and install ssh.exe in Cygwin and delete the weird one I ended up DL'ing.

**Comment by gbraad** [comment]
Windows 10's SSH beta should work. Has been verified and tested by me for
quite a while as part of the insider program.

Cygwin is still preferred at the moment, as many of the ssh clients on
Windows give weird results or unexpected output.

-- 

   Gerard Braad | http://gbraad.nl
   [ Doing Open Source Matters ]

**Comment by karlnicholas** [comment]
Right - It's not obvious in the documentation anywhere that minishift depends on an installed SSH application and since it's an executable you tend to think it has everything it needs built in. This would go well on the front page. 

Also, it might be a good idea to set the `CYGWIN=nodosfilewarning` env variable when running, if possible. I think the issue is that the proper Cygwin SSH has it defaulted to (no)dosfilewarning and the version I picked up has it defaulted to on for some reason. I don't think it can hurt anything. Perhaps I can submit a pull request but I'm not sure where to start looking for setting initial variables. Thanks for the follow-up.

Good to know about the SSH beta but I deleted it anway because it doesn't support rsa.

**Comment by gbraad** [comment]
We do not need to have SSH installed, as we can fall back to a SSH-implementation inside the minishift executable. However, as soon as an `ssh.exe` is detected on PATH, we use that instead. On Linux or macOS this 'never' causes an issue, but having a plethora of options of ssh clients on Windows creates a problem for us while testing. So, yes... we have some issues to add supported SSH clients to the documentation. it would be great to see a PR ;-)

**Comment by LalatenduMohanty** [comment]
Resolved by #1952

**Comment by parveenballyan** [comment]
Any idea? i am trying on MAC M1 silicon, how to fix this issue?




   Downloading ISO 'https://github.com/minishift/minishift-centos-iso/releases/download/v1.17.0/minishift-centos7.iso'
 375.00 MiB / 375.00 MiB [======================================================================================================================================================================] 100.00% 0s
-- Starting Minishift VM ........................... FAIL E1206 12:21:11.903897   30426 start.go:499] Error starting the VM: Error creating the VM. Error creating machine: Error in driver during machine creation: IP address never found in dhcp leases file Temporary Error: Could not find an IP address for 4a:fc:5c:1:48:11
Temporary Error: Could not find an IP address for 4a:fc:5c:1:48:11
Temporary Error: Could not find an IP address for 4a:fc:5c:1:48:11
Temporary Error: Could not find an IP address for 4a:fc:5c:1:48:11
Temporary Error: Could not find an IP address for 4a:fc:5c:1:48:11
Temporary Error: Could not find an IP address for 4a:fc:5c:1:48:11
Temporary Error: Could not find an IP address for 4a:fc:5c:1:48:11
